builders/pallets-precompiles/precompiles/randomness.md

```diff
@@ -0,0 +1,291 @@
+
+- **MAX_RANDOM_WORDS** - the maximum number of random words being requested. Set to {{ networks.moonbase.randomness.max_random_words }} words
+- **MIN_VRF_BLOCKS_DELAY** - the minimum number of blocks before a request can be fulfilled for local VRF requests. Set to {{ networks.moonbase.randomness.min_vrf_blocks_delay }} blocks
+- **MAX_VRF_BLOCKS_DELAY** - the maximum number of blocks before a request can be fulfilled for local VRF requests. Set to {{ networks.moonbase.randomness.max_vrf_blocks_delay }} blocks
+- **REQUEST_DEPOSIT_AMOUNT** - the deposit amount needed to request random words. There is one deposit per request. Set to {{ networks.moonbase.randomness.req_deposit_amount }} Wei
+
+### Events {: #events }
+
+The interface includes the following events:
+
+- **FulfillmentSucceeded**() - emitted when the request has been successfully executed
+- **FulfillmentFailed**() - emitted when the request has failed to execute fulfillment
+
+### Enums {: #enums }
+
+The interface includes the following enums:
+
+- **RequestStatus** - the status of the request, which can be `DoesNotExist` (0), `Pending` (1), `Ready` (2), or `Expired` (3)
+- **RandomnessSource** - the type of the randomness source, which can be `LocalVRF` (0) or `RelayBabeEpoch` (1)
+
+## The Randomness Consumer Solidity Interface {: #randomness-consumer-solidity-interface }
+
+The [`RandomnessConsumer.sol`](https://github.com/PureStake/moonbeam/blob/4e2a5785424be6faa01cd14e90155d9d2ec734ee/precompiles/randomness/RandomnessConsumer.sol){target=_blank} Solidity interface makes it easy for smart contracts to interact with the randomness precompile. Using the randomness consumer ensures the fulfillment comes from the randomness precompile. 
+
+The consumer interface includes the following functions:
+
+- **fulfillRandomWords**(*uint256* requestId, *uint256[] memory* randomWords) - handles the VRF response for  given request. This method is triggered by a call to `rawFulfillRandomWords`
+- **rawFulfillRandomWords**(*uint256* requestId, *uint256[] memory* randomWords) - executed when the [`fulfillRequest` function](#:~:text=fulfillRequest(uint256 requestId)) of the randomness precompile is called. The origin of the call is validated, ensuring the randomness precompile is the origin, and then the `fulfillRandomWords` method is called
+
+## Local VRF {: #local-vrf }
+
+Local VRF randomness is generated on a block-by-block basis at the beginning of the block using the previous block's VRF output along with the public key of the current block author's VRF key. The generated randomness result is stored and used to fulfill all randomness requests for the current block.
+
+If your contract could have concurrent requests open, you can use the `requestId` returned from the `requestLocalVRFRandomWords` method to track which response is associated with which randomness request. Colliding `requestId`s are cryptographically impossible as long as the salts differ.
```

Comment:
```suggestion
If your contract could have concurrent requests open, you can use the `requestId` returned from the `requestLocalVRFRandomWords` method to track which response is associated with which randomness request.
```
Colliding requestIds are unlikely because of the pallet logic -- it doesn't depend on the salt at all.